cloud/gcp/metadata.go

```diff
@@ -0,0 +1,23 @@
+package gcp
+
+import (
+	"cloud.google.com/go/compute/metadata"
+	"github.com/treeverse/lakefs/logging"
+)
+
+type MetadataProvider struct {
+	logger logging.Logger
+}
+
+func NewMetadataProvider(logger logging.Logger) *MetadataProvider {
+	return &MetadataProvider{logger: logger}
+}
+
+func (m *MetadataProvider) GetMetadata() map[string]string {
+	projectID, err := metadata.NumericProjectID()
+	if err != nil {
+		m.logger.Errorf("%v: failed to get Google numeric project ID from instance metadata", err)
```

Comment:
I think this should be a warning - it doesn't affect the health or correctness of lakeFS in any way (nor do we ever request explicit permissions to do this operation (also true for the AWS metadata provider))